app/models/proceeding.rb

```diff
@@ -78,4 +82,9 @@ def application_proceeding_type
     proceeding_type = ProceedingType.find_by ccms_code: ccms_code
     legal_aid_application.application_proceeding_types.find_by(proceeding_type_id: proceeding_type.id)
   end
+
+  def highest_proceeding_case_id
+    rec = self.class.order(proceeding_case_id: :desc).first
+    rec.nil? || rec.proceeding_case_id.nil? ? FIRST_PROCEEDING_CASE_ID : rec.proceeding_case_id
```

Comment:
I think we need to add an index on proceeding_case_id for fast lookups - might as well make it a unique index to ensure we don't get dupes

Reply:
Good point. I've pushed a commit to add an index.